### ISSUE TITLE
Non-blank value for UID required by Alerting Provisioning HTTP API for removing default contact point

### ISSUE THREAD

**Issue by dugalp** [grafana/grafana#66087]
**What happened**:

Trying to provision contact points from source code configurations and cleaning up anything users may have created to ensure consistency from code to instance.

**What you expected to happen**:

We expect to be able to delete the default contact point after updating the root notification policy to a different contact point. However, the UID of the default contact point is blank, and the API needs a non-blank one. 

**How to reproduce it (as minimally and precisely as possible)**:

Step 1: Update the root notification policy to a non-default contact point.

Step 2: List the contact-points

```
curl https://admin:pwd@grafana-hostname.companyname.com/api/v1/provisioning/contact-points

    {
        "uid": "",
        "name": "email receiver",
        "type": "email",
        "settings": {
            "addresses": "\u003cexample@email.com\u003e"
        },
        "disableResolveMessage": false
    }
```    
Step 3: remove the Contact point with a blank UID.

```
curl -X DELETE https://admin:pwd@grafana-hostname.companyname.com/api/v1/provisioning/contact-points/
{"message":"Not found"}
```
Note that the documentation mentioned a UID is required at the end of the URL, after "contact-points/{UID}" and we can't provide it as it's blank.

**Anything else we need to know?**:

The change request is here in the Pull Request:
https://github.com/grafana/grafana/pull/66089

**Environment**: Not applicable
- Grafana version: Grafana enterprise 9.4.7 
- Data source type & version: Not applicable
- OS Grafana is installed on: Not applicable
- User OS & Browser: Not applicable
- Grafana plugins: Not applicable
- Others: Not applicable

**Comment by mellieA** [comment]
I'm not able to reproduce this, can you share how you created the non-default contact that has a blank uid? 

curl admin:admin@localhost:3000/api/v1/provisioning/contact-points | json_pp
[
   {
      "disableResolveMessage" : false,
      "name" : "new-email",
      "settings" : {
         "addresses" : "some_email@serverdomain.com",
         "singleEmail" : false
      },
      "type" : "email",
      "uid" : "7V5MgSY4zz"
   },

**Comment by dugalp** [comment]
The default is not created by me. It's probably created in code of Grafana. The change request is here in the Pull Request:
https://github.com/grafana/grafana/pull/66089

I have found a workout which is to leave the default there, however, if I switch default and rebuild the Grafana instance, it will re-appear if I delete it via clickops using console.

**Comment by nikos** [comment]
I am also experiencing this with a fresh Grafana installation (9.5.8) even if I do not modify anything (like described in the original bug report step 2). So it seems with the Alerting Provisioning API v1 impossible to change the default  eamil receiver ("grafana-default-email") programmatically since the UID is an empty string.

**Comment by m4rc77** [comment]
What happened to this bug and the related PR https://github.com/grafana/grafana/pull/66089? **This issue is NOT yet solved with Grafana 11.2.2** 

It can easily be reproduced with the help of https://github.com/grafana/provisioning-alerting-examples/tree/main/config-files. 

Just start the example with `docker compose up` an then navigate to http://localhost:3000/alerting/notifications?search=&tab=contact_points and click the "Export all" button. As a result you get:

```apiVersion: 1
contactPoints:
    - orgId: 1
      name: My Contact Email Point
      receivers:
        - uid: my_contact_email_point_uid
          type: email
          settings:
            addresses: your_email@company.org
            message: '{{ template "custom_email.message" .}}'
            singleEmail: false
            subject: '{{ template "custom_email.subject" .}}'
          disableResolveMessage: false
    - orgId: 1
      name: email receiver
      receivers:
        - uid: ""
          type: email
          settings:
            addresses: <example@email.com>
          disableResolveMessage: false
```
the receiver `email receiver` with the email `example@email.com` is created even though it is not defined in `grafana/provisioning/alerting/alert_resources.yaml`. And there is also no way to overwrite it as it has no UUID!

The provisioned contact points from the example are:

```
contactPoints:
    - orgId: 1
      name: My Contact Email Point
      receivers:
        - uid: my_contact_email_point_uid
          type: email
          settings:
            addresses: your_email@company.org
            message: '{{ template "custom_email.message" .}}'
            singleEmail: false
            subject: '{{ template "custom_email.subject" .}}'
          disableResolveMessage: false
```

**In summary, there is currently (at least with 11.2.2) no way to get rid of the `example@email.com` contact point, when using alert provisioning. This is dangerous and could potentially leak sensitive internal operational information to whoever OWNs that email address!**

**Comment by sjoerdhooft** [comment]
I came across the same issue, after provisioning the contactpoints through files I ended up with a grafana-default-email contactpoint with two email addresses, one being the example@email.com, which had an uid of ""

We use helmfile to render the manifests, and I managed to get rid of that email address by adding the ```deleteContactPoints``` to my provisioning file:

```
  contactpoints.yaml: |
    apiVersion: 1
    contactPoints:
    - name: grafana-default-email
      orgId: 1
      receivers:
      - disableResolveMessage: false
        settings:
          addresses: $grafanaSmtpUser
          message: Please check Grafana for more details
          singleEmail: false
        type: email
        uid: email_default_uid
    deleteContactPoints:
    - orgId: 1
      uid: ""
```
